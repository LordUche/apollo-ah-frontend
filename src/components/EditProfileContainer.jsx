import React, {Component} from 'react';
import { connect } from 'react-redux';

import EditProfile from '../views/EditProfile';
import ProfileHeaderContainer from './ProfileHeaderContainer';
import Wrapper from '../views/reuse/Wrapper';
import {updateUserProfile} from '../actions/updateProfileAction';


/**
 * Container component for the EditProfile view
 * @export
 * @class EditProfileContainer
 * @extends {Component}
 */
  class EditProfileContainer extends Component{
     /**
     *Creates an instance of EditProfileContainer.
     * @param {object} props The props object
     * @memberof EditProfileContainer
     */
    constructor(props){
        super(props);
        this.handleUpdateProfile = this.handleUpdateProfile.bind(this);
    }

    handleUpdateProfile(event){
      event.preventDefault();
      const data = new FormData(event.target);
      const {updateProfile} = this.props;
      updateProfile(data);
    }
    
     /**
     * The render method
     * @returns {array} The resulting JSX object
     * @memberof EditProfileContainer
     */
    render(){
      const {isLoading, profile, errorData} = this.props;
        return (
          <Wrapper>
            <ProfileHeaderContainer activePage="PROFILE" />
            <EditProfile 
            handleUpdateProfile={this.handleUpdateProfile} 
            isLoading={isLoading} 
            profile={profile}
            errorData={errorData}
            />
          </Wrapper>
        );
    }
}

const mapStateToProps = state => ({
  isLoading: state.loading,
  profile: state.user,
  errorData: state.user.error,
});

const mapDispatchToProps = dispatch => ({
  updateProfile: userData => dispatch(updateUserProfile(userData)),
});
export default connect(mapStateToProps, mapDispatchToProps)(EditProfileContainer);
