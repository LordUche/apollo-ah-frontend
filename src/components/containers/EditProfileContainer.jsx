import React, { Component } from 'react';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';

import EditProfile from '../views/EditProfile';
import ProfileHeaderContainer from './ProfileHeaderContainer';
import { updateUserProfile } from '../../actions/profileAction';
import { UPDATING_PROFILE } from '../../actions/actionTypes';

/**
 * Container component for the EditProfile view
 * @export
 * @class EditProfileContainer
 * @extends {Component}
 */
class EditProfileContainer extends Component {
  /**
     *Creates an instance of EditProfileContainer.
     * @param {object} props The props object
     * @memberof EditProfileContainer
     */
  constructor(props) {
    super(props);
    this.handleUpdateProfile = this.handleUpdateProfile.bind(this);
  }

  /**
   * Updates the user profile
   * @param {object} event The event object
   * @memberof EditProfileContainer
   */
  handleUpdateProfile(event) {
    event.preventDefault();
    const data = new FormData(event.target);
    const { updateProfile } = this.props;
    updateProfile(data);
  }

  render() {
    const { isLoading, profile } = this.props;
    return (
      <>
        <ProfileHeaderContainer activePage="PROFILE" />
        <EditProfile
          handleUpdateProfile={this.handleUpdateProfile}
          isLoading={isLoading}
          profile={profile}
        />
      </>
    );
  }
}

EditProfileContainer.propTypes = {
  updateProfile: PropTypes.func.isRequired,
  isLoading: PropTypes.bool.isRequired,
  profile: PropTypes.shape({
    firstname: PropTypes.string.isRequired,
    lastname: PropTypes.string.isRequired,
    username: PropTypes.string.isRequired,
    bio: PropTypes.string.isRequired,
    errorData: PropTypes.arrayOf(PropTypes.object).isRequired,
  }).isRequired,
};


const mapStateToProps = state => ({
  isLoading: state.loading.updatingProfile,
  profile: state.user,
});

const mapDispatchToProps = dispatch => ({
  updateProfile: userData => dispatch(updateUserProfile(UPDATING_PROFILE, userData)),
});
export default connect(mapStateToProps, mapDispatchToProps)(EditProfileContainer);
