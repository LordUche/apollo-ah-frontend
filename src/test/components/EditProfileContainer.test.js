import React from 'react';
import setup from '../setup';
import EditProfileContainer from '../../components/EditProfileContainer';

const container = <EditProfileContainer activePage="PROFILE" />;

describe('<EditProfileContainer>', () => {
  it('should render without crashing', () => {
    const wrapper = setup(container);
    expect(wrapper).toMatchSnapshot();
  });
});
