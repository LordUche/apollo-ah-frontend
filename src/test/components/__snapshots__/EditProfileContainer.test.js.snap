// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditProfileContainer> should render without crashing 1`] = `ReactWrapper {}`;
